### DISCUSSION TITLE
Unable to install Linux on Surface Pro 9.

### DISCUSSION THREAD

**Discussion by Valerio1474** [linux-surface/linux-surface#1366] [Q&A] [answered]
Hello, in the past I have installed successfully Ubuntu on SP4, SP5, SP7, thanks to internet and to your guides, now I've a lot of troubles installing Ubuntu on SP9.
From what I'm reading it should be simple, but it's not. First, I have troubles booting on USB sticks. I've tried two different thumb drives, loaded with Ubuntu 22, but they are very rarely recognised. They are "old" USB-A drives, but with an USB-C adapter, they should work. I've tested them on my laptop USB-C port before attaching them to SP9 and they worked.
When I attach to SP9 port the thumb drive (on UEFI there are already "boot from USB" on top and "allow 3rd party CA" set up) Ubuntu doesn't start.
Ubuntu started from USB only once, then I was able to install it, but it booted once and never again. So I tried again to boot from USB to fix the installation, thinking something was wrong with grub, but as I told you, I was unable to boot from USB again.
In UEFI options, "boot from USB" is on top, 3rd party CA is allowed.
The internal drive is decrypted (bitlocker is disabled), Windows 11 has been updated to the latest version, so the drivers and firmwares before getting into the process of installing Linux on my SP9. 
Any hint?

**Comment by efc64** [comment]
I'm not sure whether this helps at all but I recently installed linux on my SP8 with a USB drive from 2016/17 (Kingston 101 G2)

**Comment by Valerio1474** [comment]
Hi, thanks for your reply. Has your drive an USB-C plug? I'm thinking if
the adapter to USB-C can be a problem. Otherwise it can be a firmware
problem, but in this case there is little I can do, you cannot downgrade
it, just wait for the next update.

Il Mer 7 Feb 2024, 18:03 efc64 ***@***.***> ha scritto:

**Comment by efc64** [comment]
I used a USB-C adapter too as most of my thumb drives are USB-A like yours. These are the adapters I bought, I try to avoid Amazon but it's too useful for bits like this: https://www.amazon.co.uk/dp/B07CVX3516

**Comment by Valerio1474** [comment]
Thanks, can you tell me what's your firmware version? Have you tried to
boot from USB recently?

Il Gio 8 Feb 2024, 00:54 efc64 ***@***.***> ha scritto:

**Comment by efc64** [comment]
This is what I see when I boot into UEFI:

UEFI: 23.101.143
Nuvoton: 7.2.2.0
SMF controller: 66.0.0.38
SAM controller: 11.200.139
Intel management engine: 15.0.2411.5
Tcon firmware: 161.39.33.0
PD contoller: 10.4.0
Intel type--c retimer: 2.22.0.3

I installed PopOS off USB last weekend and only checked for firmware updates when in Windows a few weeks back upon initially receiving the device, so I'm fairly certain I've been on those versions throughout.

**Comment by Valerio1474** [comment]
Thank you, sadly the version numbers differ a lot from mine, probably
numbering starts all over again with a new device. Do you have something
called Secured Core? Did you disable it to boot your USB stick?

Il Ven 9 Feb 2024, 00:02 efc64 ***@***.***> ha scritto:

**Comment by Mister-Odd** [comment]
I have a SP9 too and I also used both a USB C and a USB A with adapter..

I "think" your issue is the Secure Boot w/ 3rd party CA. Try it with just the Microsoft Key. Also - turn off "Secure Core" - I've never been able to boot with that on. 

I had no issues with Debian. Haven't tried Ubuntu. I'm currently using Arch / systemd boot secure boot. You may just want to turn off secure boot get it all configured and turn it on again once everything is setup right. 

Check out rEFInd too http://www.rodsbooks.com/refind/ it's the awesome sauce and that site's info  will make you a Boot Master Guru 5000. 

Also:
https://github.com/linux-surface/linux-surface/wiki/UEFI <-- make sure you don't have "Enable boot configuration lock" enabled.

**Reply by Valerio1474** [reply]
Hi, I already disabled this option, but you are right, the problem is in UEFI/Debian/Ubuntu, even with secure boot completely disabled they don't boot. Debian boots with the "Microsoft and third-party CA" option but I had to hack into the /EFI/Boot folder to manually add mmx64.efi, otherwise I would have kept getting the missing file error. I tried Arch and it is able to boot, unfortunately I don't know this distribution, so I would have to learn it from scratch.

**Comment by Mister-Odd** [comment]
I went from Windows 11 to Debian (Trixie) with a USB C to A adapter and I never touched the secureboot/UEFI settings. It was a total after-thought and it installed flawless into the new partition. If you have another USB drive - I'd try it with a new ISO and see if it works. I use Rufus /w the DD Image Mode setting to write my ISO files. https://github.com/pbatard/rufus (Image Mode doesn't boot for me - only DD Image Mode does).

Debian has always been my unbreakable "everything just works" rockstar distro.

Arch is a bit of a learning curve, but not a huge difference.  I find I can break stuff - clean it up - test new things - clean it up again and it always boots.. no matter how silly I've been. 

It's pacman -S vs apt install. The rest of it is 99% regular Linux commands and your DE/WM.

Ubuntu was weird for me. Not bad, again it's mostly Debian w/ a touch of different. I just found it strange in enough ways to not use it.

For secure boot on Arch the assisted process with sbctl from: https://wiki.archlinux.org/title/Unified_Extensible_Firmware_Interface/Secure_Boot#Assisted_process_with_sbctl , was crazy easy. I just copied and pasted the steps, rebooted and *poof* Arch was running under (systemd) secure boot. 

I don't see why it couldn't do the same for any distro :  https://github.com/Foxboron/sbctl

**Reply by Valerio1474** [reply]
Thank you for your guidelines and the readings. I'm going to try a lot of things in the next days.
I'm using Debian since 1999, and I forgot how much it was difficult at that time to install it (compared to Mandrake or SuSE), but on my Surface devices it was always simpler using Ubuntu than Debian (autorotation and other tablet stuff worked OOTB on Ubuntu). Now these problems are giving me some showstoppers but I'm sure some Linux distribution is going to be inside my SP9 quite soon.

**Comment by Valerio1474** [comment]
I must dig it further but maybe this is happening to me:
Surface Pro 6 - Live USB won't boot
https://github.com/linux-surface/linux-surface/issues/1335

**Reply by Valerio1474** [reply, marked as answer]
I've found that indeed it was my firmware, updated to the latest version, to cause the problem I was facing.
Solved doing this:
Create an installation USB stick for Debian and edit it inside \EFI\Boot to add a copy of grubx64.efi named mmx64.efi
disable secure boot and secure core
boot from USB and install Debian (Debian won't boot at this point)
Open UEFI and boot on Windows to fix the bootloader
Now launch a terminal as admin
list of commands:
mountvol.exe P:  /S  (the EFI partition is then mounted as  P:  drive)
cd P:
cd EFI\debian
dir (it shows a list of efi files like shimx64.efi, grubx64.efi, etc...)
copy mmx64.efi original_mmx64.efi (just in case something happens)
copy grubx64.efi mmx64.efi (you rename the copy to overwrite mmx64.efi)
reboot
Now GRUB works and you can play with Debian.
Thank you Mr-Odd and EFC64

P.S.: At the moment, Windows cannot be booted from GRUB, you need to access UEFI firmware and manually (swipe left) launch it. The error is about shim something...

**Reply by efc64** [reply]
Apologies, I was away from home for a few days but am glad you resolved it! At least you can still get to Windows, albeit through a convoluted method
